tests/click_tests/device_menu/test_check_backup.py

```
@@ -0,0 +1,105 @@
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+from typing import TYPE_CHECKING
+
+import pytest
+
+from trezorlib import messages
+
+from ... import translations as TR
+from ...common import MNEMONIC12
+from .common import Menu, assert_device_screen, close_device_menu, open_device_menu
+
+if TYPE_CHECKING:
+    from trezorlib.debuglink import DebugLink
+    from trezorlib.messages import Features
+
+    from ...device_handler import BackgroundDeviceHandler
+# Trezor Safe 7 only
+pytestmark = [pytest.mark.models("eckhart")]
+
+
+def prepare_check_backup(debug: "DebugLink", features: "Features") -> None:
+    check_backup_title = TR.reset__check_backup_title
+    assert check_backup_title in Menu.SECURITY.content(features)
+
+    # Open device menu
+    open_device_menu(debug)
+
+    # Navigate to device menu
+    Menu.SECURITY.navigate_to(debug, features)
+
+    # Trigger check backup
+    layout = debug.read_layout()
+    label_idx = layout.vertical_menu_content().index(check_backup_title)
+    debug.button_actions.navigate_to_menu_item(label_idx)
+
+
+@pytest.mark.setup_client(uninitialized=True)
+def test_uninitialized_fails(device_handler: "BackgroundDeviceHandler"):
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+    assert features.initialized is False
+    assert features.pin_protection is False
+    assert features.unfinished_backup is False
+
+    # Device is uninitialized, security settings are not accessible in the settings
+    with pytest.raises(TypeError, match="argument of type 'NoneType' is not iterable"):
```

Can we raise a specific exception type for this error (here and below)?
(IIUC, currently this test relies on `None` being iterated upon to raise a specific error)

---

Could you please also update the `pytest.raises(AssertionError, match=...)` below (to be a bit more specific)?

---

https://github.com/trezor/trezor-firmware/pull/5630/commits/0292a784d12a6c2b968db3380df4520fbafbf708